sam/utils.py

```diff
@@ -39,9 +35,17 @@
 
 
 def func_to_tool(fn: callable) -> dict:
+    """
+    Convert a function to an OpenAI tool definition based on signature and docstring.
+
+    The docstring should be formatted using the Google Napolean style.
+    """
     signature: inspect.Signature = inspect.signature(fn)
-    description, args = fn.__doc__.split("Args:")
-    doc_data = yaml.safe_load(args.split("Returns:")[0])
+    if signature.parameters:
+        description, args = fn.__doc__.split("Args:")
+        doc_data = yaml.safe_load(args.split("Returns:")[0])
+    else:
+        description = fn.__doc__
```

Comment:
`doc_data` is being used in the end of the function, but it is not defined in the else statement.